doc/gallery/plot_rasterio.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+"""
+.. _recipes.rasterio:
+
+=================================
+Parsing rasterio's geocoordinates
+=================================
+
+
+The example illustrates how to use an accessor (see :ref:`internals.accessors`)
+to  convert a projection's cartesian coordinates into 2D longitudes and
+latitudes.
+
+These new coordinates might be handy for plotting and indexing, but it should
+be kept in mind that a grid which is regular in projection coordinates will
+likely be irregular in lon/lat. It is often recommended to work in the data's
+original map projection.
+"""
+
+import os
+import urllib.request
+import numpy as np
+import xarray as xr
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+from rasterio.warp import transform
+
+
+# Define the accessor
+@xr.register_dataarray_accessor('rasterio')
+class RasterioAccessor(object):
```

Comment:
I would lean towards keeping this as just a function -- it keeps the example simpler and an accessor is probably overkill for this problem.